subsys/bluetooth/services/ias/ias_internal.h

```diff
@@ -0,0 +1,62 @@
+/** @file
+ *  @brief Internal APIs for Immediate Alert Service.
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <bluetooth/gatt.h>
+
+enum bt_ias_alert_lvl {
+	BT_IAS_ALERT_LVL_NO_ALERT,
+	BT_IAS_ALERT_LVL_MILD_ALERT,
+	BT_IAS_ALERT_LVL_HIGH_ALERT,
+};
+
+struct bt_ias_client {
+	struct bt_conn *conn;
+	struct bt_uuid_16 uuid;
+	struct bt_gatt_write_params write;
+	struct bt_gatt_discover_params discover;
+};
+
+struct bt_ias_client_cb {
+	/** @brief Callback function for bt_ias_discover.
+	 *
+	 *  This callback is called when discovery procedure is complete.
+	 *
+	 *  @param conn Bluetooth connection object.
+	 *  @param err 0 on success, ATT error or negative errno otherwise
+	 */
+	void (*discover)(struct bt_conn *conn, int err);
+};
+
+/** @brief Set alert level
+ *
+ *  @param conn Bluetooth connection object
+ *  @param alert_lvl Alert level value to be written
+ *
+ *  @return Zero in case of success and error code in case of error.
+ *
+ */
+int bt_ias_alert_write(struct bt_conn *conn, enum bt_ias_alert_lvl alert_lvl);
+
+/** @brief Discover Immediate Alert Service
+ *
+ *  @param conn Bluetooth connection object
+ *
+ *  @return Zero in case of success and error code in case of error.
+ */
+int bt_ias_discover(struct bt_conn *conn);
+
+/** @brief Register Immediate Alert Client callbacks
+ *
+ *  @param cb The callback structure
+ *
+ *  @return Zero in case of success and error code in case of error.
+ */
+int bt_ias_client_cb_register(struct bt_ias_client_cb *cb);
```

Comment:
Should these not be in a public header file?

Reply:
Yup,  sorry about that. It's all in `ias.h` now.